src/transformers/models/greaselm/feature_extraction_greaselm.py

```diff
@@ -0,0 +1,739 @@
+                    continue
+                have_stop = False
+                # remove all concepts having stopwords, including hard-grounded ones
+                for t in c.split("_"):
+                    if t in nltk_stopwords:
+                        have_stop = True
+                if not have_stop and c in self.cpnet_vocab_underscores:
+                    prune_qc.append(c)
+
+            ac = item["ac"]
+            prune_ac = []
+            for c in ac:
+                if c[-2:] == "er" and c[:-2] in ac:
+                    continue
+                if c[-1:] == "e" and c[:-1] in ac:
+                    continue
+                all_stop = True
+                for t in c.split("_"):
+                    if t not in nltk_stopwords:
+                        all_stop = False
+                if not all_stop and c in self.cpnet_vocab_underscores:
+                    prune_ac.append(c)
+
+            item["qc"] = prune_qc
+            item["ac"] = prune_ac
+
+            prune_examples.append(item)
+        return prune_examples
+
+    def ground(self, common_sense_example: Dict) -> Dict:
+
+        # common_sense_example: Dict
+        """
+        {'answerKey': 'B',
+         'id': 'b8c0a4703079cf661d7261a60a1bcbff', 'question': {'question_concept': 'magazines',
+                      'choices': [{'label': 'A', 'text': 'doctor'}, {'label': 'B', 'text': 'bookstore'},
+                                  {'label': 'C', 'text': 'market'}, {'label': 'D', 'text': 'train station'}, {'label':
+                                  'E', 'text': 'mortuary'}],
+                      'stem': 'Where would you find magazines along side many other printed works?'},
+         'statements': [
+             {'label': False, 'statement': 'Doctor would you find magazines along side many other printed works.'},
+             {'label': True, 'statement': 'Bookstore would you find magazines along side many other printed works.'},
+             {'label': False, 'statement': 'Market would you find magazines along side many other printed works.'},
+             {'label': False,
+              'statement': 'Train station would you find magazines along side many other printed works.'},
+             {'label': False, 'statement': 'Mortuary would you find magazines along side many other printed works.'}]}
+        """
+        statements = []
+        answers = []
+
+        for statement in common_sense_example["statements"]:
+            statements.append(statement["statement"])
+
+        for answer in common_sense_example["question"]["choices"]:
+            ans = answer["text"]
+            answers.append(ans)
+
+        grounded_concepts = self.match_mentioned_concepts(statements, answers)
+        res = self.prune(grounded_concepts)
+
+        return res
+
+    def concepts2adj(self, node_ids):
+        cids = np.array(node_ids, dtype=np.int32)
+        n_rel = len(self.id2relation)
+        n_node = cids.shape[0]
```

Comment:
A lot of the variable names are a bit unclear, where possible, please rename them to something more understandable.

cids => concept_ids, n_rel => number_of_relations, etc.